### PR TITLE
Duplicate WM 

### PR DESCRIPTION
**Pull request by Hitenjain14**
### Changes
-

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: zboxcore/sdk/commitworker.go [hunk 1]
```diff
@@ -143,6 +143,11 @@ func (commitreq *CommitRequest) processCommit() {
 	}
 
 	if lR.LatestWM != nil {
+		if lR.LatestWM.ClientID != client.GetClientID() {
```
Comment (lpoli) [inline zboxcore/sdk/commitworker.go]: This check seems useful for debug purpose only.
Signature verification will also prove if client id is same or not

File: zboxcore/fileref/fileref.go [hunk 2]
```diff
@@ -128,6 +128,10 @@ func (r *Ref) GetHash() string {
 	return r.Hash
 }
 
+func (r *Ref) GetHashData() string {
+	return fmt.Sprintf("%s:%s", r.AllocationID, r.Path)
```
Comment (lpoli) [inline zboxcore/fileref/fileref.go]: Lets also include fileID in hash calculation for directory.

**Review by lpoli — CHANGES_REQUESTED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Review by lpoli — APPROVED** [review]

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/create-dir  |
| gosdk  | two-phase-timestamp  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | test-createdir  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | master  |
| gosdk  | two-phase-timestamp  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | test-createdir  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Pull request merged**